sway-core/src/type_system/type_info.rs

```diff
@@ -29,6 +29,12 @@ impl fmt::Display for AbiName {
     }
 }
 
+#[derive(Debug, Clone, Hash)]
+pub struct Length {
+    pub val: usize,
```

Comment:
I have a slight preference for using `.val()` and `.span()` accessor methods, since I've done so many refactors where I wish there was a defined API 😓 but I think this is probably low-risk.  Worth considering, though.

Reply:
Reasonable suggestion. I will do that.